test/unittests/test_utils_test.cpp

```diff
@@ -70,3 +77,30 @@ TEST(test_utils, print_c_execution_result)
     str_value << FizzyExecutionResult{false, true, {42}};
     EXPECT_EQ(str_value.str(), "result(42 [0x2a])");
 }
+
+TEST(test_utils, print_typed_execution_result)
+{
+    std::stringstream str_trap;
+    str_trap << TypedExecutionResult{Trap, {}};
+    EXPECT_EQ(str_trap.str(), "trapped");
+
+    std::stringstream str_void;
+    str_void << TypedExecutionResult{Void, {}};
+    EXPECT_EQ(str_void.str(), "result()");
+
+    std::stringstream str_value_i32;
+    str_value_i32 << TypedExecutionResult{ExecutionResult{Value{42}}, ValType::i32};
```

Comment:
Does this constructor verify that the input is `<=2^32-`? If not, can you put a test with a large value?

Reply:
Here you mean the `Value{}` constructor. It currently has overloads for many integer types and maps them to wasm types. No additional check is required.